src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,70 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { useState } from 'react';
+import { User } from '../../types/User';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const [isLoading, setIsLoading] = useState(false);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
```

Comment:
Consider moving `setIsLoading(true)` inside the `if (currentTodo)` block to avoid setting the loading state when there is no current todo.